File: algorithms/2032.two-out-of-three/solution.go
func twoOutOfThree(nums1 []int, nums2 []int, nums3 []int) []int {
	res := []int{}

	for i := 1; i <= 100; i++ {
		count := 0

		if contains(nums1, i) {
			count++
		}

		if contains(nums2, i) {
			count++
		}

		if contains(nums3, i) {
			count++
		}

		if count >= 2 {
			res = append(res, i)
		}
	}

	return res
}

func contains(arr []int, x int) bool {
	for _, v := range arr {
		if v == x {
			return true
		}
	}
	return false
}

File: algorithms/2032.two-out-of-three/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_out_of_three(nums1: Vec<i32>, nums2: Vec<i32>, nums3: Vec<i32>) -> Vec<i32> {
        let mut res = vec![];

        for n in 1..=100 {
            let mut count = 0u32;

            if nums1.contains(&n) {
                count += 1;
            }

            if nums2.contains(&n) {
                count += 1;
            }

            if nums3.contains(&n) {
                count += 1;
            }

            if count >= 2 {
                res.push(n as i32);
            }
        }

        res
    }
}
